src/common.h

```diff
@@ -131,11 +131,25 @@ extern int  vsnprintfz(char *dst, size_t n, const char *fmt, va_list args);
 extern int  snprintfz(char *dst, size_t n, const char *fmt, ...) __attribute__ (( format (printf, 3, 4)));
 
 // memory allocation functions that handle failures
+#ifdef NETDATA_LOG_ALLOCATIONS
+#define strdupz(s) strdupz_int(__FILE__, __FUNCTION__, __LINE__, s)
+#define callocz(nmemb, size) callocz_int(__FILE__, __FUNCTION__, __LINE__, nmemb, size)
```

Comment:
__FUNCTION__ is non standard, you should detect it in autoconf, probably __func__ then __FUNCTION__ then fallback to define global __func__ with "", checkout if [1] helps.

[1] https://www.gnu.org/software/autoconf-archive/ax_c_var_func.html